src/prettytable/prettytable.py

```diff
@@ -596,16 +610,22 @@ def _validate_function(self, name, val):
 
     def _validate_hrules(self, name, val):
         try:
-            assert val in (ALL, FRAME, HEADER, NONE)
+            if sys.version_info >= (3, 12):
+                assert val in HRuleStyle
+            else:
+                assert val in HRuleStyle._value2member_map_
         except AssertionError:
-            msg = f"Invalid value for {name}. Must be ALL, FRAME, HEADER or NONE."
+            msg = f"Invalid value for {name}. Must be HRuleStyle."
             raise ValueError(msg)
 
     def _validate_vrules(self, name, val):
         try:
-            assert val in (ALL, FRAME, NONE)
+            if sys.version_info >= (3, 12):
+                assert val in VRuleStyle
+            else:
+                assert val in VRuleStyle._value2member_map_
```

Comment:
Also this one (and the next): `_value2member_map_` is undocumented, so is it "private" and open to change/removal with no notice?

Reply:
I don't think it get's removed for Python `<3.11`. If you like, I can change that as well though.

Reply:
I might have found an even better solution with `val in list(VRuleStyle)`. That also works for `random.choice`. See https://github.com/jazzband/prettytable/pull/327/commits/084beaefa2c12aa8704c6a66079827f5a8f89574

Reply:
Much nicer!